custom_components/econnect_metronet/coordinator.py

```diff
@@ -0,0 +1,96 @@
+        A timeout ensures the method doesn't hang indefinitely. In case of invalid token,
+        the method attempts to re-authenticate and get an update. If the update fails,
+        the connection is reset to ensure proper data alignment in subsequent runs.
+
+        Raises:
+            InvalidToken: When the token used for the connection is invalid.
+            UpdateFailed: When there's an error in updating the data.
+
+        Notes:
+            If the update fails, the method resets the device connection to ensure data
+            alignment and prevent the integration from getting stuck.
+        """
+        try:
+            if self.data is None:
+                # First update, no need to wait for changes
+                username = self.config_entry.data[CONF_USERNAME]
+                password = self.config_entry.data[CONF_PASSWORD]
+                await self.hass.async_add_executor_job(self.device.connect, username, password)
+                return await self.hass.async_add_executor_job(self.device.update)
+
+            async with async_timeout.timeout(POLLING_TIMEOUT):
+                # `device.has_updates` implements e-Connect long-polling API. This
+                # action blocks the thread for 15 seconds, or when the backend publishes an update
+                # POLLING_TIMEOUT ensures an upper bound regardless of the underlying implementation.
+                _LOGGER.debug("Coordinator | Waiting for changes (long-polling)")
+                status = await self.hass.async_add_executor_job(self.device.has_updates)
+                if status["has_changes"]:
+                    _LOGGER.debug("Coordinator | Changes detected, sending an update")
+                    return await self.hass.async_add_executor_job(self.device.update)
+                else:
+                    _LOGGER.debug("Coordinator | No changes detected")
+                    return {}
+        except InvalidToken:
+            _LOGGER.debug("Coordinator | Invalid token detected, authenticating")
+            username = self.config_entry.data[CONF_USERNAME]
+            password = self.config_entry.data[CONF_PASSWORD]
+            await self.hass.async_add_executor_job(self.device.connect, username, password)
+            _LOGGER.debug("Coordinator | Authentication completed with success")
+            return await self.hass.async_add_executor_job(self.device.update)
+        except UpdateFailed:
+            # Resetting the connection, forces an update during the next run. This is required to prevent
+            # a misalignment between the `AlarmDevice` and backend known IDs, needed to implement
+            # the long-polling strategy. If IDs are misaligned, then no updates happen and
+            # the integration remains stuck.
+            _LOGGER.debug(
+                "Coordinator | Update failed, resetting IDs to force a full update when the connection is stable"
+            )
+            self.device.reset()
```

Comment:
This code is different and changes how the connection issue is handled. Revert the change and be sure to behave as the hotfix 2.0.1. Then we may think to change the logic in the near future as a standalone PR.